convert_wide_to_long_format/convert_wide_to_long_format.py

```diff
@@ -0,0 +1,74 @@
+import os
+import argparse
+import gzip
+import re
+
+COLUMN= 'COLUMN';
+VALUE= 'VALUE';
+
+def map_row_with_header_index(row, header_index, output_file):
+    for header_row in header_index:
+        new_row_dict = {
+            key: row[idx] if isinstance(idx, int) and 0 <= idx < len(row) else idx
+            for key, idx in header_row.items()
+        }
+        output_file.write("\t".join(str(v) for v in new_row_dict.values()) + "\n")
+
+def run(input_file, output_file, custom_columns):
+    # write the array of objects to a csv file
+    if os.path.exists(output_file):
+        os.remove(output_file)  # Overwrite the file with an empty DataFrame
+        print(f"File '{output_file}' is already exist, so it is cleared first to remove the old contents")
+    # read in the files
+    try:
+        with gzip.open(output_file,"wt",encoding="utf-8") as output:
+            # Read the first line (header)
+            print('start indexing')
+            with gzip.open(input_file, 'rt') as input:
+                match_dict = {}
+                resulted_index_header = []
+                header = input.readline().strip().split('\t')
+                for column in header:
+                    if column ==  'FINNGENID':
+                        continue
+                    else:
+                        pattern = fr"^{column}_({custom_columns})?$"
+                        default_dict = {}
+                        default_dict[header[0]] = 0
+                        default_dict[COLUMN] = column
+                        default_dict[VALUE] = header.index(column)
+                        match_dict = {name: idx for idx, name in enumerate(header) if re.match(pattern, name)}
+                        if (len(match_dict) > 0):
+                            combined_dict = {**default_dict, **match_dict}
+                            print('.', end='', flush=True)
+                            new_data = {key.replace(f"{column}_", ''): value for key, value in combined_dict.items()}
+                            resulted_index_header.append(new_data)
+                print('Finished indexing and reading row data!')
+                first_keys = resulted_index_header[0].keys()
```

Comment:
hashmap keys are not guaranteed to be in inserted order so can mess up column ordering. NEed a way to ensure order stays the same.... could enumerate in order of   pheno,pheno_suffix1,pheno_suffix2 and the suffixes would be in the order given on command line